docarray/array/storage/milvus/backend.py

```diff
@@ -0,0 +1,320 @@
+        )
+        offset = FieldSchema(
+            name='offset', dtype=DataType.VARCHAR, max_length=1024, is_primary=True
+        )
+        dummy_vector = FieldSchema(
+            name='dummy_vector', dtype=DataType.FLOAT_VECTOR, dim=1
+        )
+
+        schema = CollectionSchema(
+            fields=[offset, document_id, dummy_vector],
+            description='offset2id for DocumentArray',
+        )
+
+        return Collection(
+            name=self._config.collection_name + '_offset2id',
+            schema=schema,
+            using=self._connection_alias,
+            # **self._config.collection_config,  # we probably don't want to apply the same config here
+        )
+
+    def _ensure_unique_config(
+        self,
+        config_root: dict,
+        config_subindex: dict,
+        config_joined: dict,
+        subindex_name: str,
+    ) -> dict:
+        if 'collection_name' not in config_subindex:
+            config_joined['collection_name'] = _sanitize_collection_name(
+                config_joined['collection_name'] + '_subindex_' + subindex_name
+            )
+        return config_joined
+
+    def _doc_to_milvus_payload(self, doc):
+        return self._docs_to_milvus_payload([doc])
+
+    def _docs_to_milvus_payload(self, docs: 'Iterable[Document]'):
+        extra_columns = [
+            [self._map_column(doc.tags.get(col), col_type) for doc in docs]
+            for col, col_type in self._config.columns.items()
+        ]
+        return [
+            [doc.id for doc in docs],
+            [self._map_embedding(doc.embedding) for doc in docs],
+            [doc.to_base64(**self._config.serialize_config) for doc in docs],
+            *extra_columns,
+        ]
+
+    @staticmethod
+    def _docs_from_query_response(response):
+        return DocumentArray([Document.from_base64(d['serialized']) for d in response])
+
+    @staticmethod
+    def _docs_from_search_response(
+        responses,
+    ) -> 'Union[List[DocumentArray], DocumentArray]':
+        das = []
+        for r in responses:
+            das.append(
+                DocumentArray(
+                    [Document.from_base64(hit.entity.get('serialized')) for hit in r]
+                )
+            )
+        return das if len(das) > 0 else das[0]
+
+    def _update_consistency_level(self, **kwargs):
+        kwargs_consistency_level = kwargs.get('consistency_level', None)
+        kwargs['consistency_level'] = (
+            kwargs_consistency_level
+            if kwargs_consistency_level
+            else self._config.consistency_level
+        )
+        return kwargs
+
+    def _map_embedding(self, embedding):
+        if embedding is not None:
+            from docarray.math.ndarray import to_numpy_array
+
+            embedding = to_numpy_array(embedding)
+
+            if embedding.ndim > 1:
+                embedding = np.asarray(embedding).squeeze()
+        else:
+            embedding = np.zeros(self._config.n_dim)
```

Comment:
I think for some distances (maybe cosine), this needs epsilon added or to be random, can we check ?

Reply:
this is not an issue